### PR TITLE
Update deps to latest version

### PR DESCRIPTION
**Pull request by j0k3r**
Add Node 12 & 14 in Travis.

Here are the missing one (for now):

```
$ npm outdated
Package                 Current  Wanted  Latest  Location
eslint                   5.16.0  5.16.0  7.22.0  bestzip
eslint-config-prettier    4.3.0   4.3.0   8.1.0  bestzip
prettier                 1.19.1  1.19.1   2.2.1  bestzip
```

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,6 +1,9 @@
 language: node_js
+
 node_js:
 - 10
+- 12
+- 14
 - lts/*
```
Comment (nfriedly) [inline .travis.yml:7]: The current LTS release is 14 right now, so these two are identical.

IMO, 12 is also unnecessary. If it passes in node 10, then it's not using any newer APIs, and if it passes in LTS and stable, then it's not using any deprecated APIs. I find it unlikely that a breaking change would be rolled out to node 12 without but not 10 or 14 or 15.

Comment (j0k3r) [reply .travis.yml:7]: Ok I'll revert these changes

File: package.json [hunk 2]
```diff
@@ -21,24 +21,24 @@
     "lib/"
   ],
   "dependencies": {
-    "archiver": "^4.0.2",
+    "archiver": "^5.3.0",
```
[the remark this answers is not in the document]
Comment (j0k3r) [reply package.json:24]: - Dropped Node < 10
- Use of `readdir-glob` instead of `glob` be memory efficient

And some deps updated

**Comment by nfriedly** [comment]
Hey @j0k3r, thanks for this! 

I think it's mostly good, although since `bestzip` is pretty heavily dependent on `archiver`, I do want to know if whatever constituted a breaking change in `archiver` should also be published as a breaking change here.

`eslint` and friends can also be updated, although that might necessitate an increase in the minimum node.js version that we test against. ESLint tends to be pretty aggressive about adopting new node.js features.

**Comment by j0k3r** [comment]
The only breaking change from `archiver` is https://github.com/archiverjs/node-archiver/pull/433:
> absolute glob patterns are not supported: `archiver.glob('*.txt',{cwd: '/tmp'})` has to be used instead of `archiver.glob('/tmp/*.txt')`

But I don't think it affects bestzip.

About ESLint, for now there are sill allowing Node 10:

> Prerequisites: Node.js (^10.12.0, or >=12.0.0)

I'll work on upgrading ESLint and their friends too

File: .eslintrc.js [hunk 3]
```diff
@@ -1,14 +1,19 @@
...
+    es6: true,
+  },
+  extends: ["eslint:recommended", "plugin:prettier/recommended"],
+  rules: {
+    strict: ["error", "global"],
+    "prefer-const": "error",
+    "prettier/prettier": [
+      "error",
+      {
+        endOfLine: "auto",
```
Comment (nfriedly) [inline .eslintrc.js:15]: TIL! 'auto' seems fine for now.

**Review by nfriedly — APPROVED** [review]
Nice work, thanks!

**Comment by nfriedly** [comment]
Should be published as v2.2.0 shortly

**Pull request merged**